### PR TITLE
WT-11654 Label setup commands in evergreen to ensure failures are bucketized correctly

### PR DESCRIPTION
**Pull request by ruby-oujo**
I've gone through the `evergreen.yml` file and labelled tasks with type "`setup`" that are run to setup the environment for our testing, external tasks that are relied upon for specific tests, and also the post-test tasks such as upload. 

I've compiled a list of ambiguous tasks that I am not sure if we want to classify as "setup":

- Compile mongo (this is external to us)
- Compile / make wiredtiger (the failure still shows up - but do we consider this part of the setup process, or compile in itself a test) 
- python config check (checking the python binary matches cmake)
- compile / update wiredtiger docs or similar to adjusting documentation
- build and push antithesis container

Please feel free to review the current tasks, or mention any I have missed. Thanks.

### CODE REVIEW

**Review by mickgraham-mongodb — APPROVED** [review]
LGTM

**Comment by ajmorton** [comment]
As discussed the current changes look good. For the ambiguous cases:

>    Compile mongo (this is external to us)

This should be a test since previous steps copy the latest WiredTiger code into Mongo before compilation. This can catch errors if we break the Mongo-WiredTiger API

>    Compile / make wiredtiger (the failure still shows up - but do we consider this part of the setup process, or compile in itself a test) 

Test since this will report any compiler errors we may have accidentally introduced

>    python config check (checking the python binary matches cmake)

setup since this is checking the platform config

>    compile / update wiredtiger docs or similar to adjusting documentation

Compile docs is a test since our changes could have broken the docs. Update docs is setup since the work is all shell commands like rsync and git push.

>    build and push antithesis container

This is also setup. We compile WiredTiger in previous steps so any failures will in this task will come from antithesis/docker

**Comment by mickgraham-mongodb** [comment]
Sorry I completely missed the ambiguous cases question. I agree with @ajmorton. The concern on compile is if we have a timeout, which we see in MacOS. That is clearing a setup/platform issue but if its a compile error that is from our code. I don't think we have the control we need on those cases so best left as a test.

**Comment by ruby-oujo** [comment]
Thanks for the input and review - I have added the 3 additional tasks here: [116a1f1](https://github.com/wiredtiger/wiredtiger/pull/9662/commits/116a1f1a92cb356762d52cc1183d428cf7da9afa)

File: test/evergreen.yml [hunk 1]
```diff
@@ -544,6 +559,7 @@ functions:
         remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${task_name}_${build_id}${postfix|}.tgz
     # We remove the wiredtiger directory here to avoid to getting archived again by post tasks.
     - command: shell.exec
+      type: setup
```
Comment (ajmorton) [inline test/evergreen.yml:562]: On re-read this needs to be `test`. When we run cppsuite above the command always succeeds and saves the exit code to file. If cppsuite fails this is the command that will detect and report it.

Comment (ruby-oujo) [reply test/evergreen.yml]: updated here [e83fa92](https://github.com/wiredtiger/wiredtiger/pull/9662/commits/e83fa924a0362168d2d671df5d49245ffea45907)

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks!

**Comment by ruby-oujo** [comment]
evergreen merge

**Pull request merged**